modules/wso2bps/templates/3.5.1/repository/conf/axis2/axis2.xml.erb

```diff
@@ -0,0 +1,747 @@
+           tracing & keeping track of the time at which a particular response was sent.
+
+           NOTE: This should be the very last phase in this flow
+        -->
+        <phase name="MsgOutObservation"/>
+        <!--Following phase is added to publish stats -->
+        <phase name="StatReporting"/>
+    </phaseOrder>
+
+    <clustering class="org.wso2.carbon.core.clustering.hazelcast.HazelcastClusteringAgent"
+                enable="<%= @clustering['enabled'] %>">
+
+        <!--
+           This parameter indicates whether the cluster has to be automatically initalized
+           when the AxisConfiguration is built. If set to "true" the initialization will not be
+           done at that stage, and some other party will have to explictly initialize the cluster.
+        -->
+        <parameter name="AvoidInitiation">true</parameter>
+
+        <!--
+           The membership scheme used in this setup. The only values supported at the moment are
+           "multicast" and "wka"
+
+           1. multicast - membership is automatically discovered using multicasting
+           2. wka - Well-Known Address based multicasting. Membership is discovered with the help
+                    of one or more nodes running at a Well-Known Address. New members joining a
+                    cluster will first connect to a well-known node, register with the well-known node
+                    and get the membership list from it. When new members join, one of the well-known
+                    nodes will notify the others in the group. When a member leaves the cluster or
+                    is deemed to have left the cluster, it will be detected by the Group Membership
+                    Service (GMS) using a TCP ping mechanism.
+        -->
+        <parameter name="membershipScheme"><%= @clustering['membership_scheme'] %></parameter>
+
+<%- if @clustering['membership_scheme'] == 'private-paas' -%>
+        <!-- private-paas membership scheme specific parameters -->
+        <parameter name="membershipSchemeClassName">org.wso2.carbon.ppaas.PrivatePaaSBasedMembershipScheme</parameter>
+        <parameter name="clusterIds"><%= @clustering['private_paas'] %></parameter>
+<%- end -%>
+
+<%- if @clustering['membership_scheme'] == 'kubernetes' -%>
+        <!-- kubernetes membership scheme specific parameters -->
+        <parameter name="membershipSchemeClassName">org.wso2.carbon.membership.scheme.kubernetes.KubernetesMembershipScheme</parameter>
+        <parameter name="KUBERNETES_MASTER"><%= @clustering['kubernetes']['kubernetes_master'] %></parameter>
+        <parameter name="KUBERNETES_NAMESPACE"><%= @clustering['kubernetes']['kubernetes_namespace'] %></parameter>
+        <parameter name="KUBERNETES_SERVICES"><%= @clustering['kubernetes']['kubernetes_services'] %></parameter>
+        <parameter name="KUBERNETES_MASTER_USERNAME"><%= @clustering['kubernetes']['kubernetes_master_username'] %></parameter>
```

Comment:
We don't need `KUBERNETES_MASTER`, `KUBERNETES_MASTER_USERNAME` or `KUBERNETES_MASTER_PASSWORD`